### PR TITLE
Add Anonymizable Interfaces to keep track of the anonymization process

### PR DESCRIPTION
**Pull request by RocKordier**

### CODE REVIEW

File: src/Visitor/GraphNavigator.php [hunk 1]
```diff
@@ -54,6 +56,12 @@ public function visit($value): void
 
     private function visitObject($value): void
     {
+        if ($value instanceof AnonymizableInterface) {
```
Comment (Baachi) [inline src/Visitor/GraphNavigator.php]: Wouldn't be better to have this as extension?

Comment (Baachi) [inline src/Visitor/GraphNavigator.php]: Wouldn't it be better to have this as an extension/event subscriber?

File: src/Visitor/GraphNavigator.php [hunk 2]
```diff
@@ -74,6 +78,10 @@ private function visitObject($value): void
 
         $this->dispatcher->dispatch(new PostAnonymizeEvent($value));
 
+        if ($value instanceof BeenAnonymizedInterface) {
+            $value->beenAnonymized();
```
Comment (Baachi) [inline src/Visitor/GraphNavigator.php:94]: I like the method names a lot! :+1:

**Review by Baachi — CHANGES_REQUESTED** [review]
Hey, thanks for this PR!

First of all, I like the method names. But wouldn't it be better to have this as an event subscriber?

File: src/Visitor/GraphNavigator.php (hunk 1 above)
Comment (RocKordier) [reply src/Visitor/GraphNavigator.php]: As described I don't see any other options to guarantee that the functions will be executed as first and last one if it's part of the event system.  Either `stopPropagation` or the order of priorities would prevent that.
Likewise, there is no other place where every object is checked

**Comment by RocKordier** [comment]
@Baachi  see comments above

File: src/BeenAnonymizedInterface.php [hunk 3]
```diff
@@ -0,0 +1,10 @@
+<?php
+
+declare(strict_types=1);
+
+namespace PHPSharkTank\Anonymizer;
+
+interface BeenAnonymizedInterface
+{
+    public function beenAnonymized(): void;
```
Comment (Baachi) [inline src/BeenAnonymizedInterface.php]: Maybe `hasBeenAnonymized` would be a better name which sounds more what it should do.

**Pull request merged**